### PR TITLE
Login splash page

### PR DESCRIPTION
**Pull request by diegocepedaw**
Adds the option to enable a login splash screen that will require users to log in.

### CODE REVIEW

File: src/oncall/ui/static/js/loginsplash.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+var oncall = {
```
Comment (dwang159) [inline src/oncall/ui/static/js/loginsplash.js]: maybe change the name to `oncallLogin` or something to be more distinct from `oncall.js`

File: src/oncall/ui/templates/loginsplash.html [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+<html>
+<head>
+<meta name="viewport" content="width=device-width, initial-scale=1">
+{% assets "css_libs" %} <link href="{{ ASSET_URL }}" rel="stylesheet"> {% endassets %}
+{% assets "loginsplash_css" %} <link href="{{ ASSET_URL }}" rel="stylesheet"> {% endassets %}
+<title>Oncall</title>
+</head>
+<body>
+
+<div id="id01" class="modal">
```
Comment (dwang159) [inline src/oncall/ui/templates/loginsplash.html]: Give a more descriptive id, or remove if unused

File: src/oncall/ui/static/css/loginsplash.css [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    height: 100%; /* Full height */
+    overflow: auto; /* Enable scroll if needed */
+    background-color: rgb(0,0,0); /* Fallback color */
+    background-color: rgba(0,0,0,0.4); /* Black w/ opacity */
+    padding-top: 60px;
+}
+
+/* Modal Content/Box */
+.modal-content {
+    background-color: #fefefe;
```
Comment (dwang159) [inline src/oncall/ui/static/css/loginsplash.css]: Change to #F6F8FA to be consistent with the rest of the site's background. Also, I kind of like width 30% here, with `.container` having width 80%. Not a strong opinion, but I prefer the splash screen with a small form that Iris has.

**Comment by dwang159** [comment]
Linking to #165. Hopefully should be sufficient to address this use-case.

File: src/oncall/ui/static/css/loginsplash.css (hunk 3 above)
Comment (diegocepedaw) [reply src/oncall/ui/static/css/loginsplash.css]: I implemented all the requested changes.

File: configs/config.yaml [hunk 4]
```diff
@@ -178,6 +178,9 @@ iris_plan_integration:
 # Configures whether Slack settings will appear in the frontend
 slack_instance: foobar
 
+# setting to toggle login splash screen. This will require users to log in to  be able to use the app. Set to True to enable and False to disable.
+login_splash: False
```
Comment (shoxv) [inline configs/config.yaml]: should we rename this property to something like force_login?

File: src/oncall/ui/static/js/loginsplash.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    csrfKey: 'csrf-key'
+  },
+  callbacks: {
+    onLogin: function (data){
+      // callback for successful user login. Reloads page to go to the normal oncall interface.
+      location.reload();
+    }
+  },
+  init: function(){
+    var self = this;
```
Comment (shoxv) [inline src/oncall/ui/static/js/loginsplash.js]: not needed

File: src/oncall/ui/static/js/loginsplash.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    this.events.call(this);
```
Comment (shoxv) [inline src/oncall/ui/static/js/loginsplash.js]: dont need to use .call here. you can just do this.events()

**Pull request merged**